src/medley/core.cljc

```diff
@@ -357,6 +357,64 @@
   ([pred coll]
    (rest (drop-while (complement pred) coll))))
 
+(defn partition-after
+  "Returns a lazy sequence of partitions, splitting after `(pred item)` returns
+  true. Returns a transducer when no collection is provided."
+  ([pred]
+   (fn [rf]
+     (let [v (volatile! [])
+           split? (volatile! false)]
+       (fn
+         ([] (rf))
+         ([result]
+          (rf (if-some [v (not-empty @v)]
+                (unreduced (rf result v))
+                result)))
+         ([result x]
+          (if @split?
+            (let [result (rf result @v)]
+              (vreset! v (if (reduced? result) [] [x]))
+              (when-not (pred x) (vreset! split? false))
+              result)
+            (do (when (pred x) (vreset! split? true))
+                (vswap! v conj x)
+                result)))))))
+  ([pred coll]
+   (lazy-seq
+    (when-let [s (seq coll)]
+      (let [run (take-upto pred s)]
+        (cons run (partition-after pred (lazy-seq (drop (count run) s)))))))))
+
+(defn partition-before
+  "Returns a lazy sequence of partitions, splitting before `(pred item)` returns
+  true. Returns a transducer when no collection is provided."
+  ([pred]
+   (fn [rf]
+     (let [v (volatile! [])]
```

Comment:
Maybe instead of `v`, `part` would be more descriptive?

Reply:
Done